lib/src/over_react_redux/over_react_redux.dart

```diff
@@ -0,0 +1,393 @@
+import 'package:over_react/src/component_declaration/builder_helpers.dart' as builder_helpers;
+import 'package:collection/collection.dart';
+import 'package:js/js.dart';
+import 'package:over_react/over_react.dart';
+import 'package:react/react_client.dart';
+import 'package:react/react_client/react_interop.dart';
+import 'package:react/react_client/js_backed_map.dart';
+import 'package:redux/redux.dart';
+
+
+
+part 'over_react_redux.over_react.g.dart';
+
+@PropsMixin(keyNamespace: '')
+abstract class _$ConnectPropsMixin {
+  Map get props;
+  void Function(dynamic action) dispatch;
+}
+
+/// A wrapper around the JS react-redux `connect` function that supports OverReact components.
+///
+/// __Example:__
+///
+///     UiFactory<CounterProps> ConnectedCounter = connect<CounterState, CounterProps>(
+///         mapStateToProps: (state) => (
+///           Counter()..intProp = state.count
+///         ),
+///         mapDispatchToProps: (dispatch) => (
+///           Counter()..increment = () => dispatch(INCREMENT_ACTION())
+///         ),
+///     )(Counter);
+///
+/// - [mapStateToProps] is used for selecting the part of the data from the store that the connected
+///  component needs.
+///  - It is called every time the store state changes.
+///  - It receives the entire store state, and should return an object of data this component needs.
+/// If you need access to the props provided to the connected component you can use [mapStateToPropsWithOwnProps],
+/// the second argument will be `ownProps`.
+/// See: <https://react-redux.js.org/using-react-redux/connect-mapstate#defining-mapstatetoprops>
+///
+/// - [mapDispatchToProps] will be called with dispatch as the first argument.
+/// You will normally make use of this by returning new functions that call dispatch() inside themselves,
+/// and either pass in a plain action directly or pass in the result of an action creator.
+/// If you need access to the props provided to the connected component you can use [mapDispatchToPropsWithOwnProps],
+/// the second argument will be `ownProps`.
+/// See: <https://react-redux.js.org/using-react-redux/connect-mapdispatch#defining-mapdispatchtoprops-as-a-function>
+///
+/// - [mergeProps] if specified, defines how the final props for the wrapped component are determined.
+/// If you do not provide [mergeProps], the wrapped component receives {...ownProps, ...stateProps, ...dispatchProps}
+/// by default.
+///
+/// - [areStatesEqual] does a simple `==` check by default.
+/// - [areOwnPropsEqual] does a shallow Map equality check by default.
+/// - [areStatePropsEqual] does a shallow Map equality check by default.
+/// - [areMergedPropsEqual] does a shallow Map equality check by default.
+///
+/// - [context] can be utilized to provide a custom context object created with `createContext`.
+/// [context] is how you can utilize multiple stores. If you must, it is not reccomended. :P
```

Comment:
> If you must, it is not reccomended. :P

I'm fine with the snark, but grammatically this sentence is a little confusing. What about something like:

> While supported, this is not recommended. :P

A reference to this might be nice, too